packages/teams-js/src/public/appNotification.ts

```diff
@@ -0,0 +1,98 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Namespace to interact with the appNotifications specific part of the SDK
+ */
+export namespace appNotification {
+  /**
+   * Data structure to represent appNotification information
+   */
+  export interface NotificationDisplayParam {
+    /**The heading of the notiification to be delivered,it would have a maximum string length of 20*/
+    title: string;
+    /**the body of information which would be displayed in the notification box,it would have a maximum string length of 40*/
+    Content: string;
+    /**this would represent an icon that can be displayed on the notification*/
+    icon?: HTMLImageElement;
```

Comment:
>HTMLImageElement [](http://example.com/codeflow?start=11&length=16)

Can you say more about how you chose this type for this property? Since this icon is really just an image, it seems like it would be extra and unnecessary work for a developer to create an HTML DOM `img` element to hold the icon. Moreover, I'm not sure how passing a DOM element like this across the iframe boundary could work. The IMG element will be created inside the app's IFRAME, and it couldn't really be access by the host SDK or the hosting app. Have you considered just using either a URL to the image or the bytes of the image? Are there any restrictions on its size?